### PR TITLE
fix: sidebar closing on link click

### LINKED ISSUE #196: pop-up navigation menu closes even when control-clicking a link in it

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #196.

### CODE REVIEW

File: components/sections/Navigation/index.tsx [hunk 1]
```diff
@@ -33,11 +33,10 @@ const Navigation = () => {
 			>
 				<div className='flex flex-col gap-4 xl:flex-row xl:gap-8'>
 					<Link
-						className='nav-link-shadow transition-all text-sky-300'
+						className='nav-link-shadow text-sky-300 transition-all'
 						href='/cch'
 						onClick={() => {
 							trackEvent('homepage_mainnav_cch')
-							setOpen(false)
```
Comment (hamirmahal) [inline components/sections/Navigation/index.tsx:40]: I'm not sure why clicking on a link in the pop-up menu should have closed it. I might be missing something here.

Comment (jonaro00) [reply components/sections/Navigation/index.tsx:40]: Wouldn't you want the menu to close and let you read the page you clicked on?
The only time you'd want it to stay open is when Ctrl-clicking as in https://github.com/shuttle-hq/www/issues/196#issuecomment-1837116890 right? If so, this onClick handler might be able to look for [ctrlKey](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/ctrlKey) and close the menu based on it's value.

Comment (hamirmahal) [reply components/sections/Navigation/index.tsx:40]: Ah, I see.

**Comment by hamirmahal** [comment]
How do you feel about extracting some of the duplicate logic in the `Link`s into some sort of `LinkWrapper` component in a follow-up pull request?

**Review by jonaro00 — APPROVED** [review]
Perhaps. I'll leave that to @GugaGongadze.

(LGTM; There is just one conflict to resolve)

**Comment by hamirmahal** [comment]
Okay, I resolved the conflict and did some slight refactoring.

**Comment by hamirmahal** [comment]
test

**Comment by hamirmahal** [comment]
What changes would you like to see in this pull request before merging it?

**Review by GugaGongadze — CHANGES_REQUESTED** [review]
Thanks for the PR, @hamirmahal.

Just tested it on my Mac machine and it does not work, it is due to Command key on mac not matching `ctrlKey`, `metaKey` will match though. So we need a combination of those 2 to make it work.

**Comment by hamirmahal** [comment]
You're welcome, @GugaGongadze.

**Comment by hamirmahal** [comment]
The latest changes seem to have the desired behavior on a macOS device.

https://github.com/shuttle-hq/www/assets/43425812/1cbf757a-c1aa-48e6-a00e-f69a937fad1a

**Comment by hamirmahal** [comment]
I'm not sure why [Vercel's preview](https://github.com/shuttle-hq/www/pull/197#issuecomment-1921137939) hasn't updated.

**Comment by hamirmahal** [comment]
That preview doesn't have the latest changes, so macOS functionality still may not work there, but the changes work locally.

**Review by GugaGongadze — APPROVED** [review]
Thanks, LGTM!

**Comment by hamirmahal** [comment]
You're welcome!

**Pull request merged**